File: python/LeetCode_1582_SpecialPositionsinaBinaryMatrix.py
def numSpecial(mat) -> int:
    m = len(mat)
    n = len(mat[0])
    rows = [0 for i in range(m)]
    cols = [0 for i in range(n)]
    ones = []
    for i in range(m):
        for j in range(n):
            if mat[i][j] == 1:
                rows[i] += 1
                cols[j] += 1
                ones.append((i, j))
    
    res = 0
    for x in ones:
        i, j = x
        if rows[i] == 1 and cols[j] == 1:
            res += 1

    return res


numSpecial([[1,0,0],[0,0,1],[1,0,0]])


"""
def numSpecial(mat) -> int:
    m = len(mat)
    n = len(mat[0]) 
    total = 0
    for i in range(m):
        for j in range(n):
            if (mat[i][j] == 1):
                sum = 0
                for x in range(m):
                    sum += mat[x][j]
                
                for y in range(n):
                    sum += mat[i][y]

                if sum == 2:
                    total += 1
    return total       
"""

File: c/LeetCode_1582_SpecialPositionsinaBinaryMatrix.c
#include <stdio.h>
#include <stdlib.h>


int numSpecial(int** mat, int matSize, int* matColSize) {
    int *rows = (int*) calloc(matSize, sizeof(int));
    int *columns = (int*) calloc(*matColSize, sizeof(int));
    int *ones = (int*) malloc((matSize * *matColSize) * 10 * sizeof(int));

    int inx = 0;
    for (int i = 0; i < matSize; ++i) {
        for (int j = 0; j < *matColSize; ++j) {
            if (mat[i][j] == 1) {
                rows[i] += 1;
                columns[j] += 1;

                ones[inx] = i;
                ones[++inx] = j;
                ++inx;
            }
        }
    }

    int result = 0;
    for (int i = 0; i < (inx / 2); ++i) {
        int index = 2 * i;
        int x = ones[index];
        int y = ones[index + 1];

        if (rows[x] == 1 && columns[y] == 1) {
            ++result;
        }
    }


    return result;
}


int main() {
    int matr[10][10] = {{0,0,1,0},{0,0,0,0},{0,0,0,0},{0,1,0,0}};

    int p = 4;
    numSpecial(matr, 4, &p);

    return 0;
}